src/facade/dragonfly_connection.cc

```diff
@@ -1071,6 +1071,8 @@ void Connection::EnsureAsyncMemoryBudget() {
 }
 
 std::string Connection::RemoteEndpointStr() const {
+  if (socket_ == nullptr)
```

Comment:
@romange added this a temporary workaround for the tests issue, I don't know what the proper fix would be...

Reply:
Please make RemoteEndpointStr() a virtual method, I think it's cleaner